lib/postgrex/type_module.ex

```diff
@@ -920,6 +1021,22 @@ defmodule Postgrex.TypeModule do
     end
   end
 
+  defp split_super_decode({:->, _, [head, body]}) do
+    case head do
+      [{:when, _, [pattern, sub_oids, sub_types, guard]}] ->
```

Comment:
Super extensions are private, so we can update all of them to receive the mod and simplify the logic here.